source/grid/grid_tools_cache.cc

```diff
@@ -146,6 +151,38 @@ namespace GridTools
       }
     return vertex_kdtree;
   }
+
+
+
+  template <int dim, int spacedim>
+  const RTree<std::pair<BoundingBox<spacedim>, unsigned int>> &
+  Cache<dim, spacedim>::get_covering_rtree() const
+  {
+    if (update_flags & update_covering_rtree)
+      {
+        // TO DO: Using a single bounding box per process works,
+        // but it's sub-optimal as the locally owned portion of
+        // the domain might consists of two separate pieces,
+        // resulting in an oversized bounding box.
+        // In the future some flag and a function should avoid
+        // this by splitting "big" bounding boxes.
+        const auto tria_mpi =
+          dynamic_cast<const parallel::Triangulation<dim, spacedim> *>(
+            &(*tria));
+        Assert(tria_mpi != nullptr, ExcNotImplemented());
```

Comment:
shouldn't the calls below all work now even with a single MPI process or a sequential triangulation?

Reply:
Unfortunately no: I am using the triangulation's mpi communicator, and the function `get_communicator()` is implemented only for `parallel::Triangulation`...

Reply:
But you can still use a dummy communicator like `MPI_COMM_SELF` (defined in `dealii/include/deal.II/base/mpi.h:37`) in case the `dynamic_cast` fails.

Reply:
Ok, thank you! I didn't know about it!